quic-client/src/lib.rs

```diff
@@ -41,24 +46,11 @@ pub enum QuicClientError {
 }
 
 pub struct QuicPool {
-    connections: Vec<Arc<Quic>>,
+    connections: Vec<Arc<Box<dyn BaseClientConnection>>>,
```

Comment:
Arcs are pointers, so I don't think it's necessary to have Arc<Box<dyn...>> it should just be possible to have Arc<dyn...>

Reply:
Right, fixed. Good catch!